packages/grafana-data/src/types/panel.ts

```diff
@@ -137,7 +137,9 @@ export interface PanelEditorProps<T = any> {
 /**
  * Called when a panel is first loaded with current panel model
  */
-export type PanelMigrationHandler<TOptions = any> = (panel: PanelModel<TOptions>) => Partial<TOptions>;
+export type PanelMigrationHandler<TOptions = any> = (
+  panel: PanelModel<TOptions>
+) => Partial<TOptions> | Promise<Partial<TOptions>>;
```

Comment:
```suggestion
export type PanelMigrationHandler<TOptions = unknown> = (
  panel: PanelModel<TOptions>
) => Partial<TOptions> | Promise<Partial<TOptions>>;
```

Reply:
Is it worth also doing this change?

Reply:
I initially tried to do something like this but:
 - it's a breaking (type) change!
 - it's acutally unrelated to this PR, which i want to keep small and contained to reduce the chance of revert.

This return type `any | Promise<any>` is pretty pointless for the caller PanelModel as the type is just simplified down to `any`, meaning there's no check to ensure the promise is awaited.

The other problem here is allowing for `Partial<TOptions>` to be returned (instead of just `TOptions`), when this value is assigned directly to the panel's options. It should be the same type.